pandas/io/json/normalize.py

```diff
@@ -233,31 +257,11 @@ def _recursive_extract(data, path, seen_meta, level=0):
 
                 _recursive_extract(obj[path[0]], path[1:],
                                    seen_meta, level=level + 1)
-        else:
+        elif isinstance(data, list):
```

Comment:
Related to below comment on test I don’t think this is the right approach as it doesn’t need to always be a list

Reply:
`data` can be a list or not.
This branch handles the case when `data` is a list. The `else` branch handles the case when it is not.